core/templates/services/state-interaction-stats.service.ts

```diff
@@ -99,37 +98,41 @@ export class StateInteractionStatsService {
    * Returns a promise which will provide details of the given state's
    * answer-statistics.
    */
-  computeStats(state: State): Promise<IStateRulesStats> {
+  computeStats(
+      state: State, useCachedResults: boolean): Promise<IStateRulesStats> {
```

Comment:
Hmmm, when you come across a call to this method somewhere in the code you will see something like `computeStats(state, false)` and in order to understand what does the second arg actually means you have to find this declaration. Wouldn't it make sense to have this as a private method and then have a public methods `forceComputeStats()` and `computeStats()` that would call the private method?

Reply:
Yeah good point, I decided it's not worth having the extra functionality so I've updated it to *always* cache.